geom/type_line_string.go

```diff
@@ -305,6 +305,36 @@ func (s LineString) Length() float64 {
 	return sum
 }
 
+// Centroid gives the centroid of the coordinates of the line string.
+// It returns true iff the centroid is well defined.
+func (s LineString) Centroid() (Point, bool) {
+	if s.NumPoints() == 0 {
```

Comment:
Looks like this if statement isn't necessary (that case would be handled by `sumLength == 0`).

Reply:
Fixed (removed).